components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/notification/mgt/dao/impl/NotificationDAOImpl.java

```diff
@@ -177,6 +260,85 @@ public int updateNotificationStatus(int notificationId, Notification.Status stat
         return notifications;
     }
 
+
+    @Override
+    public int getNotificationCountByStatus(Notification.Status status, int tenantId) throws NotificationManagementException {
+        int notificationCountByStatus = 0;
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        try {
+            conn = NotificationManagementDAOFactory.getConnection();
+
+            String sql = "SELECT COUNT(n1.NOTIFICATION_ID) AS NOTIFICATION_COUNT FROM " +
+                    "DM_DEVICE d, DM_DEVICE_TYPE t, (SELECT NOTIFICATION_ID, DEVICE_ID, " +
+                    "OPERATION_ID, STATUS, DESCRIPTION FROM DM_NOTIFICATION WHERE " +
+                    "TENANT_ID = ? AND STATUS = ?) n1 WHERE n1.DEVICE_ID = d.ID AND d.DEVICE_TYPE_ID=t.ID " +
+                    "AND TENANT_ID = ?";
+
+            stmt = conn.prepareStatement(sql);
+            stmt.setInt(1, tenantId);
+            stmt.setString(2, status.toString());
+            stmt.setInt(3, tenantId);
+
+            rs = stmt.executeQuery();
+            if (rs.next()) {
+                notificationCountByStatus = rs.getInt("NOTIFICATION_COUNT");
+            }
+        } catch (SQLException e) {
+            throw new NotificationManagementException("Error occurred while retrieving information of all " +
+                    "notifications", e);
+        } finally {
+            NotificationDAOUtil.cleanupResources(stmt, rs);
+        }
+        return notificationCountByStatus;
+    }
+
+    @Override
+    public List<Notification> getNotificationsByStatus(PaginationRequest request, Notification.Status status, int tenantId) throws
+            NotificationManagementException{
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet rs = null;
+        List<Notification> notifications = null;
+        try {
+            conn = NotificationManagementDAOFactory.getConnection();
+            String sql = "SELECT n1.NOTIFICATION_ID, n1.DEVICE_ID, n1.OPERATION_ID, n1.STATUS," +
+                    " n1.DESCRIPTION, d.DEVICE_IDENTIFICATION, t.NAME AS DEVICE_TYPE FROM " +
+                    "DM_DEVICE d, DM_DEVICE_TYPE t, (SELECT NOTIFICATION_ID, DEVICE_ID, " +
+                    "OPERATION_ID, STATUS, DESCRIPTION FROM DM_NOTIFICATION WHERE " +
+                    "TENANT_ID = ? AND STATUS = ?) n1 WHERE n1.DEVICE_ID = d.ID AND d.DEVICE_TYPE_ID=t.ID " +
+                    "AND TENANT_ID = ?";
+
+            sql = sql + " LIMIT ?,?";
+
+            stmt = conn.prepareStatement(sql);
+            stmt.setInt(1, tenantId);
+            stmt.setString(2, status.toString());
+            stmt.setInt(3, tenantId);
+
+            int paramIdx = 4;
+
+            stmt.setInt(paramIdx++, request.getStartIndex());
```

Comment:
Why do we need the variable paramIdx here? We can just put the respective the param index as line#316,#317, etc. Why this is special-cased as this?